java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/plugin/local/AbstractObservableLocalReader.java

```diff
@@ -140,6 +152,130 @@ public AbstractObservableLocalReader(String pluginName, String readerName) {
         super(pluginName, readerName);
     }
 
+    /**
+     * Add a {@link ObservableReader.ReaderObserver}.
+     * <p>
+     * The observer will receive all the events produced by this reader (se insertion, removal,
+     * etc.)
+     *
+     * @param observer the observer object
+     */
+    @Override
+    public final void addObserver(ObservableReader.ReaderObserver observer) {
+        if (observer == null) {
+            return;
+        }
+
+        if (logger.isTraceEnabled()) {
+            logger.trace("Adding '{}' as an observer of '{}'.", observer.getClass().getSimpleName(),
+                    getName());
+        }
+
+        synchronized (sync) {
+            if (observers == null) {
+                observers = new HashSet<ObservableReader.ReaderObserver>(1);
+            }
+            observers.add(observer);
+        }
+    }
+
+    /**
+     * Remove a {@link ObservableReader.ReaderObserver}.
+     * <p>
+     * The observer will do not receive any of the events produced by this reader.
+     *
+     * @param observer the observer object
+     */
+    @Override
+    public final void removeObserver(ObservableReader.ReaderObserver observer) {
```

Comment:
in removeObserver(ObservableReader.ReaderReaderObserver observer), the parameter isn't defined as final